app/src/androidTest/java/org/wikipedia/base/ColorAssertions.kt

```diff
@@ -0,0 +1,57 @@
+package org.wikipedia.base
+
+import android.content.Context
+import android.graphics.drawable.ColorDrawable
+import android.util.TypedValue
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.core.widget.ImageViewCompat
+import androidx.test.espresso.ViewAssertion
+import com.google.android.material.imageview.ShapeableImageView
+import org.junit.Assert.assertTrue
+
+object ColorAssertions {
+    sealed class ColorType {
+        data object TextColor : ColorType()
+        data object BackgroundColor : ColorType()
+        data object Tint : ColorType()
+        data object ShapeableImageViewColor : ColorType()
+    }
+
+    fun hasColor(
+        colorResOrAttr: Int,
```

Comment:
An overall note on color assertions:
I wonder if, instead of asserting the `R.attr` value of the expected color, we should assert the actual `R.color` expected value?
If we assert the `R.attr` value, that's basically a tautology. It's like saying "is the color I'm expecting to be Black actually expected to be Black "?
Whereas, what we should be asserting is "is the color I'm expecting to be Black **actually Black**?"

Reply:
fixed.